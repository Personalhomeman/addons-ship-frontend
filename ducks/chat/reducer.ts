import { Message, ChatActionTypes } from './types';
import { ChatActions } from './actions';

export interface ChatState {
  messages: Message[];
}

const initialState: ChatState = {
  messages: []
};

export const reducer = (state = initialState, action: ChatActions): ChatState => {
  switch (action.type) {
    case ChatActionTypes.SendMessage:
    case ChatActionTypes.ReceiveMessage:
      return {
        ...state,
        messages: [...state.messages, action.payload]
      };
    default:
      return state;
  }
};
