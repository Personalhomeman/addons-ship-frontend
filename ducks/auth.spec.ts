jest.mock('@/services/ship-api');

import configureMockStore, { MockStoreCreator, MockStoreEnhanced } from 'redux-mock-store';
import thunk from 'redux-thunk';

import shipApi from '@/services/ship-api';

import reducer, { setToken } from './auth';

describe('auth', () => {
  describe('reducer', () => {
    it('sets the token', () => {
      const state = reducer(undefined, setToken.set('some-token'));

      expect(state).toMatchSnapshot();
    });
  });

  describe('setToken', () => {
    it('fetches an app version', async () => {
      const mockStore: MockStoreCreator = configureMockStore([thunk.withExtraArgument({ shipApi })]);
      const store: MockStoreEnhanced = mockStore();

      const token = 'tokenke';

      await store.dispatch(setToken(token) as any);

      expect(store.getActions()).toMatchSnapshot();
      expect(shipApi.setToken).toHaveBeenCalledWith(token);
    });
  });
});
