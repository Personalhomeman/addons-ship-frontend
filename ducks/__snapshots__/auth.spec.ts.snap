// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth reducer sets the token 1`] = `
Object {
  "token": "some-token",
}
`;
